core/src/main/java/com/linecorp/armeria/server/AnnotatedHttpServiceConfigurator.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableList.Builder;
+
+import com.linecorp.armeria.internal.annotation.AnnotatedHttpService;
+import com.linecorp.armeria.server.annotation.ExceptionHandlerFunction;
+import com.linecorp.armeria.server.annotation.RequestConverterFunction;
+import com.linecorp.armeria.server.annotation.ResponseConverterFunction;
+
+/**
+ * A configurator to configure {@link ExceptionHandlerFunction}s, {@link RequestConverterFunction}s
+ * or {@link ResponseConverterFunction}s for an {@link AnnotatedHttpService}.
+ */
+final class AnnotatedHttpServiceConfigurator {
+
+    /**
+     * Creates a new instance with the specified {@code exceptionHandlersAndConverters}.
+     */
+    static AnnotatedHttpServiceConfigurator ofExceptionHandlersAndConverters(
+            Iterable<?> exceptionHandlersAndConverters) {
+
+        Builder<ExceptionHandlerFunction> exceptionHandlers = null;
+        Builder<RequestConverterFunction> requestConverters = null;
+        Builder<ResponseConverterFunction> responseConverters = null;
+
+        for (final Object o : exceptionHandlersAndConverters) {
+            boolean added = false;
```

Comment:
nit: How about removing this `added` flag and refactor the following logic to `if ... else if ...`. 
IMHO, it is more natural for me and there is no case that an object is instance of `ExceptionHandlerFunction` and `RequestConverterFunction`. 😉

```java
if (o instanceof ExceptionHandlerFunction) {
    ...
} else if (o instanceof RequestConverterFunction) {
    ...
} else if (o instanceof ResponseConverterFunction) {
    ...
} else {
    throw new IllegalArgumentException(...);
}
```

Reply:
Fixed it. f9d29c0